apis/Google.Cloud.Diagnostics.AspNetCore/Google.Cloud.Diagnostics.AspNetCore/Logging/GoogleLogger.cs

```diff
@@ -0,0 +1,94 @@
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Cloud.Logging.V2;
+using Google.Cloud.Diagnostics.Common;
+using Google.Protobuf.WellKnownTypes;
+using Google.Api.Gax;
+using Microsoft.Extensions.Logging;
+using System;
+using Google.Api;
+
+namespace Google.Cloud.Diagnostics.AspNetCore
+{
+    /// <summary>
+    /// <see cref="ILogger"/> for Stackdriver Logging.
+    /// </summary>
+    public class GoogleLogger : ILogger
+    {
+        /// <summary>The consumer to push logs to.</summary>
+        private readonly IConsumer<LogEntry> _consumer;
+
+        /// <summary>The minimum log level.</summary>
+        private readonly LogLevel _logLevel;
+
+        /// <summary>The formatted log name for the project and log.</summary>
+        private readonly string _logName;
+
+        /// <summary>The global resource. See: https://cloud.google.com/logging/docs/api/v2/resource-list </summary>
+        private static readonly MonitoredResource _globalResource = new MonitoredResource { Type = "global" };
+
+        internal GoogleLogger(IConsumer<LogEntry> consumer, LogLevel logLevel, string projectId, string logName)
+        {
+            GaxPreconditions.CheckNotNull(projectId, nameof(projectId));
+            GaxPreconditions.CheckNotNull(logName, nameof(logName));
+            _consumer = GaxPreconditions.CheckNotNull(consumer, nameof(consumer));
+            _logLevel = logLevel;
+            _logName = LogUtils.GetLogName(projectId, logName);
+        }
+
+        /// <summary>
+        /// Currently unsupported, always return null.
+        /// </summary>
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            // TODO(talarico): Implement.
+            return null;
+        }
+
+        /// <inheritdoc />
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            return logLevel >= _logLevel;
+        }
+
+        /// <inheritdoc />
+        public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Exception exception, Func<TState, Exception, string> formatter)
+        {
+            GaxPreconditions.CheckNotNull(formatter, nameof(formatter));
+
+            if (!IsEnabled(logLevel))
+            {
+                return;
+            }
+
+            string message = formatter(state, exception);
+            if (string.IsNullOrEmpty(message))
+            {
+                return;
+            }
+
+            LogEntry entry = new LogEntry
+            {   
+                Resource = _globalResource,
+                LogName = _logName,
```

Comment:
It's possible this will change type to be a LogNameOneofName (or whatever it's called) at some point, btw.